animatch/Screens/UIKitDownloadPage/DownloadPageViewController.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  DownloadPageViewController.swift
+//  animatch
+//
+//  Created by Ксения Панкратова on 13.01.2025.
+//
+
+import UIKit
+import SwiftUI
+
+final class DownloadPageViewController: UIViewController {
+
+    private let viewModel: DownloadPageInput
+    private let startScreenViewModel: StartScreenViewModel
+
+    private let logo = UIHostingController(rootView: NGGLogoView())
+
+    private let logInButton = UIButton()
+    private let registrationButton = UIButton()
+
+    init(viewModel: DownloadPageInput, startScreenViewModel: StartScreenViewModel) {
+        self.viewModel = viewModel
+        self.startScreenViewModel = startScreenViewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+        setupLayout()
+    }
+
+    // MARK: - UI
+
+    private func setupUI() {
+        viewModel.setStartScreenViewModel(startScreenViewModel)
```

Comment:
Тогда тут уже это всё не надо
```suggestion
    private let viewModel: DownloadPageInput

    private let logo = UIHostingController(rootView: NGGLogoView())

    private let logInButton = UIButton()
    private let registrationButton = UIButton()

    init(viewModel: DownloadPageInput) {
        self.viewModel = viewModel
        super.init(nibName: nil, bundle: nil)
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    // MARK: - Lifecycle

    override func viewDidLoad() {
        super.viewDidLoad()
        setupUI()
        setupLayout()
    }

    // MARK: - UI

    private func setupUI() {
```

Reply:
Эх, гитхаб ужасно подсветил изменения. В общем просто выпили тут весь startScreenViewModel